server/bleep/src/webserver/answer.rs

```diff
@@ -350,6 +366,27 @@ impl<'s> AnswerAPIClient<'s> {
             .send()
             .await
     }
+
+    async fn send_until_success(
+        &self,
+        prompt: &str,
+        max_tokens: u32,
+    ) -> Result<reqwest::Response, AnswerAPIError> {
+        let mut attempt = 0;
+        loop {
+            if attempt >= self.max_attempts {
+                break Err(AnswerAPIError::MaxAttemptsReached(self.max_attempts));
+            }
+            let response = self.send(prompt, max_tokens).await;
+            match response {
+                Ok(r) if r.status() == StatusCode::OK => break Ok(r),
+                Err(e) => break Err(AnswerAPIError::Fatal(e)),
+                _ => (),
+            };
+            warn!(%attempt, "answer-api returned {} ... retrying", response.unwrap().status());
+            attempt += 1;
+        }
```

Comment:
Why not use a ` for attempt in 0..self.max_attempts { .. }`  loop here?

Reply:
sure, that would clean the extra `if` check.